3.evaluate_model/scripts/nbconverted/get_LOIO_probabilities.py

```diff
@@ -0,0 +1,168 @@
+#!/usr/bin/env python
+# coding: utf-8
+
+# In[1]:
+
+
+import pathlib
+import warnings
+
+import pandas as pd
+
+from sklearn.linear_model import LogisticRegression
+from sklearn.utils import parallel_backend
+from sklearn.exceptions import ConvergenceWarning
+from joblib import load
+
+import sys
+
+sys.path.append("../utils")
+from split_utils import get_features_data
+from train_utils import get_dataset, get_X_y_data
+
+
+# In[2]:
+
+
+# load labeled data
+labeled_data_path = pathlib.Path("../0.download_data/data/labeled_data.csv.gz")
+labeled_data = get_features_data(labeled_data_path)
+
+# preview labeled data
+print(labeled_data.shape)
+labeled_data.head(5)
+
+
+# In[3]:
+
+
+# see number of images to
+num_images = labeled_data["Metadata_DNA"].unique().shape[0]
+print(f"There are {num_images} images to perform LOIO evaluation on per model.")
+
+
+# In[4]:
+
+
+# directory to load the models from
+models_dir = pathlib.Path("../2.train_model/models/")
+
+# use a list to keep track of LOIO probabilities in tidy long format for each model combination
+compiled_LOIO_wide_data = []
+
+count = 0
+
+# iterate through each model (final model, shuffled baseline model, etc)
+# sorted so final models are loaded before shuffled_baseline
+for model_path in sorted(models_dir.iterdir()):
+    # only perform LOIO with hyper params from final models
+    if "shuffled" in model_path.name:
+        continue
+
+    # load the model
+    model = load(model_path)
+    # determine feature type from model file name
+    feature_type = model_path.name.split("__")[1].replace(".joblib", "")
+
+    print(
+        f"Performing LOIO for feature type {feature_type} with parameters C: {model.C}, l1_ratio: {model.l1_ratio}"
+    )
+
+    # iterate through image paths
+    for image_path in labeled_data["Metadata_DNA"].unique():
+        # get training and testing cells from image path
+        # every cell from the image path is for testing, the rest are for training
+        train_cells = labeled_data.loc[labeled_data["Metadata_DNA"] != image_path]
+        test_cells = labeled_data.loc[labeled_data["Metadata_DNA"] == image_path]
+
+        # get X, y from training and testing cells
+        X_train, y_train = get_X_y_data(train_cells, feature_type)
+        X_test, y_test = get_X_y_data(test_cells, feature_type)
+
+        # capture convergence warning from sklearn
+        # this warning does not affect the model but takes up lots of space in the output
```

Comment:
These comments appear to apply to the next context manager (`warnings.catch_warnings()`). Consider moving these below and adding additional comments to describe the approach with parallel_backend (if warranted).

Reply:
Added some more comments to describe the approach with `parallel_backend` in https://github.com/WayScience/phenotypic_profiling_model/pull/23/commits/0ca85f5f0e452c6651eeb2d8d24464bb24058b62